File: client/src/app/modules/tiers/services/refund.type.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { RefundType } from '../../tiers/model/RefundType';

@Injectable({
  providedIn: 'root'
})
export class RefundTypeService extends AppRestService<RefundType>{

  constructor(http: HttpClient) {
    super(http, "tiers");
  }

  getRefundTypes() {
    return this.getListCached(new HttpParams(), "refund-types");
  }

  addOrUpdateRefundType(refundType: RefundType) {
    this.clearListCache(new HttpParams(), "refund-types");
    return this.addOrUpdate(new HttpParams(), "refund-type", refundType, "Enregistré", "Erreur lors de l'enregistrement");
  }

}

File: src/main/java/com/jss/osiris/modules/tiers/service/RefundTypeService.java
package com.jss.osiris.modules.tiers.service;

import java.util.List;

import com.jss.osiris.modules.tiers.model.RefundType;

public interface RefundTypeService {
    public List<RefundType> getRefundTypes();

    public RefundType getRefundType(Integer id);
	
	 public RefundType addOrUpdateRefundType(RefundType refundType);
}
